DFC.Api.Content.UnitTests/ExecuteHttpTriggerTests.cs

```diff
@@ -0,0 +1,134 @@
+            A.CallTo(() => _contentTypeMapConfig.CurrentValue).Returns(new ContentTypeMapSettings
+            {
+                OverrideUri = false,
+                ContentTypeMap = new Dictionary<string, string>() { { "test1", "Test2" }, { "test2", "Test3" } }
+            }); ;
+
+            _log = A.Fake<ILogger>();
+            _graphDatabase = A.Fake<IGraphDatabase>();
+
+            //A.CallTo(() => _neo4JHelper.GetResultSummaryAsync()).Returns(_resultSummary);
+
+            _executeFunction = new Execute(_contentTypeMapConfig, _graphDatabase);
+        }
+
+        [Fact]
+        public async Task Execute_WhenNoParametersPresent_ReturnsBadRequestObjectResult()
+        {
+            var result = await RunFunction("", null);
+
+            var badRequestObjectResult = result as BadRequestObjectResult;
+
+            // Assert
+            Assert.IsAssignableFrom<IActionResult>(result);
+            Assert.True(result is BadRequestObjectResult);
+            Assert.Equal((int?)HttpStatusCode.BadRequest, badRequestObjectResult.StatusCode);
+        }
+
+        [Fact]
+        public async Task Execute_WhenContentTypeNotPresentInMap_ReturnsBadRequestObjectResult()
+        {
+            var result = await RunFunction("abcdefghi", null);
+
+            var badRequestObjectResult = result as BadRequestObjectResult;
+
+            // Assert
+            Assert.IsAssignableFrom<IActionResult>(result);
+            Assert.True(result is BadRequestObjectResult);
+            Assert.Equal((int?)HttpStatusCode.BadRequest, badRequestObjectResult.StatusCode);
+        }
+
+        [Fact]
+        public async Task Execute_WhenContentTypePresentInMap_NoGraphData_ReturnsNotFoundObjectResultResult()
+        {
+            var result = await RunFunction("test1", null);
+
+            var notFoundObjectResult = result as NotFoundObjectResult;
+
+            // Assert
+            Assert.IsAssignableFrom<IActionResult>(result);
+            Assert.True(result is NotFoundObjectResult);
+            Assert.Equal((int?)HttpStatusCode.NotFound, notFoundObjectResult.StatusCode);
+        }
+
+        [Fact]
+        public async Task Execute_GetAllJobProfiles_ReturnsCorrectJsonResponse()
+        {
+            var recordJson = File.ReadAllText(Directory.GetCurrentDirectory() + "/Files/JobProfileRecordResponse_1.json");
+
+            A.CallTo(() => _graphDatabase.Run(A<GenericCypherQuery>.Ignored)).Returns(new List<IRecord>() { new Api.Content.UnitTests.Models.Record(new string[] { "data.properties" }, new object[] { JsonConvert.DeserializeObject<Dictionary<string, object>>(recordJson) }) });
+
+            var result = await RunFunction("test1", null);
+            var okObjectResult = result as OkObjectResult;
+
+            // Assert
+            Assert.True(result is OkObjectResult);
+
+            var resultJson = okObjectResult.Value.ToString();
+
+            var equal = JToken.DeepEquals(JToken.Parse(recordJson), JToken.Parse(resultJson));
```

Comment:
CompareNetObjects is good for this sort of thing, because you can include _what_ was different in the assert failure message.

Reply:
Received this error with CompareNetObjects so left as JToken for now:
https://github.com/GregFinzer/Compare-Net-Objects/issues/103